### ISSUE TITLE
Useful output and examples in docs

### ISSUE THREAD

**Issue by SamRoberts** [SamRoberts/fuzzy_rust#5]
When I tried to demo the original Scala project to some teammates, the first question asked was some examples ... which, of course, I didn't have!

I need to change the output of the tool so it is nicer to use. I wonder if it is possible to implement something that is a valid diff format. (I'm not sure what exactly that would mean for this project, given that you might be able to patch the text but not the pattern ... probably? Something to investigate ...)

I need to settle on a nice output format, then add some examples uses to the intro docs.

**Comment by SamRoberts** [comment]
I've merged the output change, which will be good enough for now.

It's not the standard diff format, as my tool currently produces character-by-character diffs and can have multiple diffs per line.

As part of this change, I added small technical examples of problems and corresponding scores/traces to `lib.rs`, but I have not added human meaningful examples in docs. Still need to figure out the most "rusty" way of doing this.

**Comment by SamRoberts** [comment]
I've merged another change to allow us to refer to files in the CLI.

I've decided to work on supporting arbitrary character classes. regex_syntax already supports the syntax, so I was hoping it wouldn't be hard. That said, it turned out I couldn't add "class" information to `Patt` while it was a `Copy` type, so I've removed `Copy` from `Patt` and `Text` now ... none of the `Problem` information is copied when solving now.
